core/com.b2international.snowowl.core/src/com/b2international/snowowl/core/validation/issue/ValidationIssueSearchRequest.java

```diff
@@ -175,10 +175,11 @@ protected Expression prepareQuery(ServiceProvider context) {
 		}
 		
 		if (containsKey(OptionKey.DETAILS)) {
-			final Collection<ValidationIssueDetailExtension> validationDetailExtensions = ValidationIssueDetailExtensionProvider.INSTANCE.getExtensions();
-			for (ValidationIssueDetailExtension extension : validationDetailExtensions) {
+			final Collection<String> toolingIds = getCollection(OptionKey.TOOLING_ID, String.class);
+			for (String toolingId : toolingIds) {
+				final ValidationIssueDetailExtension validationDetailExtension = ValidationIssueDetailExtensionProvider.INSTANCE.getExtensions(toolingId);
 				Options options = getOptions(OptionKey.DETAILS);
-				extension.prepareQuery(queryBuilder, options);
+				validationDetailExtension.prepareQuery(queryBuilder, options);
```

Comment:
If there are multiple `toolingId` values given, and one adds their own query clauses to the builder, we might end up with a final query that is "impossible".

Reply:
Fixed.